pyppeteer/domworld.py

```diff
@@ -0,0 +1,574 @@
+                f"{selectorOrXpath}" \
+                f"{' to be hidden' if hidden else ''}"
+        predicate = """
+        function predicate(selectorOrXPath, isXPath, waitForVisible, waitForHidden) {
+          const node = isXPath
+            ? document.evaluate(selectorOrXPath, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue
+            : document.querySelector(selectorOrXPath);
+          if (!node)
+            return waitForHidden;
+          if (!waitForVisible && !waitForHidden)
+            return node;
+          const element = /** @type {Element} */ (node.nodeType === Node.TEXT_NODE ? node.parentElement : node);
+
+          const style = window.getComputedStyle(element);
+          const isVisible = style && style.visibility !== 'hidden' && hasVisibleBoundingBox();
+          const success = (waitForVisible === isVisible || waitForHidden === !isVisible);
+          return success ? node : null;
+
+          /**
+           * @return {boolean}
+           */
+          function hasVisibleBoundingBox() {
+            const rect = element.getBoundingClientRect();
+            return !!(rect.top || rect.bottom || rect.width || rect.height);
+          }
+        }
+        """
+        waitTask = WaitTask(self, predicate, title, polling, timeout, selectorOrXpath, isXPath, visible, hidden)
+        handle = await waitTask.promise
+        if not handle.asElement():
+            await handle.dispose()
+            return None
+        return handle.asElement()
+
+
+class WaitTask(object):
+    """WaitTask class.
+
+    Instance of this class is awaitable.
+    """
+
+    def __init__(
+            self,
+            domWorld: DOMWorld,
+            predicateBody: str,
+            title: str,
+            polling: Union[str, int],
+            timeout: float,
+            loop: asyncio.AbstractEventLoop,
+            *args: Any
+    ) -> None:
+        if isinstance(polling, str):
+            if polling not in ['raf', 'mutation']:
+                raise ValueError(f'Unknown polling: {polling}')
+        elif isinstance(polling, (int, float)):
+            if polling <= 0:
+                raise ValueError(
+                    f'Cannot poll with non-positive interval: {polling}'
+                )
+        else:
+            raise ValueError(f'Unknown polling option: {polling}')
+
+        self._domWorld = domWorld
+        self._polling = polling
+        self._timeout = timeout
+        self._loop = loop
+        if args or helper.is_jsfunc(predicateBody):
+            self._predicateBody = f'return ({predicateBody})(...args)'
+        else:
+            self._predicateBody = f'return {predicateBody}'
+        self._args = args
+        self._runCount = 0
+        self._terminated = False
+        self._timeoutError = False
+        domWorld._waitTasks.add(self)
+
+        self.promise = self._loop.create_future()
+
+        async def timer(timeout: Union[int, float]) -> None:
```

Comment:
float should be enough ;)
```suggestion
        async def timer(timeout: float) -> None:
```